common/views/components/StatusIndicator/StatusIndicator.js

```diff
@@ -5,11 +5,12 @@ import Icon from '../Icon/Icon';
 
 type Props = {|
   start: Date,
-  end: Date
+  end: Date,
+  statusOverride?: ?string
 |}
 
-const StatusIndicator = ({ start, end }: Props) => {
-  const {color, text} = formatDateRangeWithMessage({start, end});
+const StatusIndicator = ({ start, end, statusOverride }: Props) => {
+  const {color, text} = formatDateRangeWithMessage({start, end, statusOverride});
```

Comment:
This feels like we're overloading this function.
`formatDateRangeWithMessage` is an abstraction but we're making it do a lot of bespoke things.

Could we not just do the override in the component?

Also thinking - that method seems to be specifically for the `StatusIndicator`, which we should use instead of using that function no? Could we move it into here?

Reply:
Yeah, that makes sense. There are a few places that aren't using the statusIndicator component, which is why I think I changed the formatDateRangeWithMessage, but I'll do what you suggest and make sure we use this component everywhere.